python/pydantic_core/_pydantic_core.pyi

```diff
@@ -822,54 +822,171 @@ class ValidationError(ValueError):
+
     def message(self) -> str: ...
+    """The formatted message associated with the error. This presents as the message template with context variables appropriately injected."""
 
 @final
 class PydanticOmit(Exception):
+    """An exception to signal that a field should be omitted from generated JSON schema.
+
+    This exception can be raised from custom JSON schema generation functions to indicate that
+    types that don't have supported JSON schema representations should be omitted from the schema,
+    rather than causing in a JSON schema build failure.
+
+    For an example, see the [customizing JSON schema](https://docs.pydantic.dev/latest/concepts/json_schema/#customizing-the-json-schema-generation-process) docs.
+    """
+
     def __new__(cls) -> Self: ...
 
 @final
 class PydanticUseDefault(Exception):
+    """An exception to signal that standard validation either failed or should be skipped, and the default value should be used instead.
+
+    This warning can be raised in custom valiation functions to redirect the flow of validation.
+    For a contextualized example, see the [validating partial json data](https://docs.pydantic.dev/latest/concepts/json/#partial-json-parsing) section of the Pydantic documentation.
+    """
+
     def __new__(cls) -> Self: ...
 
 @final
 class PydanticSerializationError(ValueError):
+    """An error raised when an issue occurs during serialization.
+
+    In custom serializers, this error can be used to indicate that serialization has failed.
+    """
+
     def __new__(cls, message: str) -> Self: ...
+    """Create a new `PydanticSerializationError`.
+
+    Arguments:
+        message: The error message associated with the serialization issue.
+    """
 
 @final
 class PydanticSerializationUnexpectedValue(ValueError):
+    """An error raised when an unexpected value is encountered during serialization.
+
+    This error is often caught and coerced into a warning, as `pydantic-core` generally makes a best attempt
+    at serializing values, in contrast with validation where errors are eagerly raised.
+
+    This is often used internally in `pydantic-core` when unexpected types are encountered during serialization,
+    but it can also be used by users in custom serializers. For example, see
+    [this example](https://github.com/pydantic/pydantic/blob/018a4555388db3a37386ed98ac6aacb48b516ede/pydantic/_internal/_generate_schema.py#L525-L530)
+    of a custom serializer for ip address like types.
+
+    Args:
+        message: The error message.
+    """
+
     def __new__(cls, message: str | None = None) -> Self: ...
+    """Create a new `PydanticSerializationUnexpectedValue` error.
+
+    Arguments:
+        message: The message associated with the unexpected value.
+    """
 
 @final
 class ArgsKwargs:
+    """An internal construct used to store arguments and keyword arguments for a function call.
```

Comment:
This is not just internal, it's used when you use a before model validator on data classes.

We should add an example of how it's used then.

Reply:
This is functionally equivalent of a namedtuple of args and kwargs.

Reply:
I've added some documentation of how `ArgsKwargs` is used in dataclass validators, as I think that's the most relevant for users.